app/src/main/java/de/westnordost/streetcomplete/quests/barrier_opening/AddBarrierOpening.kt

```diff
@@ -0,0 +1,79 @@
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.CAR
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.PEDESTRIAN
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.WHEELCHAIR
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.quests.width.WidthAnswer
+import de.westnordost.streetcomplete.screens.measure.ArSupportChecker
+
+class AddBarrierOpening(
+        private val checkArSupport: ArSupportChecker
+    ) : OsmElementQuestType<WidthAnswer> {
+
+    private val nodeFilter by lazy { """
+       nodes with
+         barrier ~ ${GATEWAYS.joinToString("|")}
+         and (!maxwidth:physical or source:maxwidth_physical ~ ".*estimat.*")
+         and (!width or source:width ~ ".*estimat.*")
+         and (!maxwidth or source:maxwidth ~ ".*estimat.*")
+         and access !~ private|no|customers|agricultural
+    """.toElementFilterExpression() }
+
+    private val excludedWaysFilter by lazy { """
+        ways with
+          highway and access ~ private|no
+    """.toElementFilterExpression() }
+
+    override val changesetComment = "Specify width of opening"
+    override val wikiLink = "Key:barrier"
+ // icon needed
+    override val icon = R.drawable.ic_quest_bicycleway_width
+    override val achievements = listOf(BICYCLIST, CAR, PEDESTRIAN, WHEELCHAIR)
+    override val defaultDisabledMessage: Int
+        get() = if (!checkArSupport()) R.string.default_disabled_msg_no_ar else 0
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_barrier_opening_width_gate
+
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> {
+        mapData.nodes.filter { nodeFilter.matches(it) }
+        val excludedWayNodeIds = mapData.ways
+            .filter { excludedWaysFilter.matches(it) }
+            .flatMapTo(HashSet()) { it.nodeIds }
+
+        return mapData.nodes
+            .filter { nodeFilter.matches(it) && it.id !in excludedWayNodeIds }
+    }
+
+
+    override fun isApplicableTo(element: Element) =
+        nodeFilter.matches(element)
+
+    override fun createForm() = AddGateWidthForm()
+
+    override fun applyAnswerTo(answer: WidthAnswer, tags: Tags, geometry: ElementGeometry, timestampEdited: Long) {
+        val key = if (tags["barrier"] in GATEWAYS) "maxwidth:physical" else "width"
+
+        tags[key] = answer.width.toOsmValue()
+
+        if (answer.isARMeasurement) {
+        tags["source:$key"] = "ARCore"
+        } else {
+        tags.remove("source:$key")
+        }
+
+        // update width:barrier if it is set
+        if (key == "width" && tags.containsKey("maxwidth")) {
+            tags["maxwidth:physical"] = answer.width.toOsmValue()
+        }
+    }
+}
+
+val GATEWAYS = setOf("gate", "entrance", "sliding_gate", "swing_gate", "wicket_gate", "bollard", "block")
```

Comment:
Please remove `bollard` and `block`, both don't fit the question and will lead to confusion (What to measure, left or right of bollard? What if there are several bollards? The smallest distance or the biggest distance between the bollards? Up to where off the path should be measured - many paths don't have clear boundaries.)

Reply:
Updated - bollard and block removed.